src/Microsoft.Extensions.Logging.EventSource/LoggingEventSource.cs

```diff
@@ -0,0 +1,234 @@
+    ///             var args = new Dictionary&lt;string, string&gt;() { { "FilterSpecs", "App*:Information;*" } };
+    ///             // Set the default level (verbosity) to Error, and only ask for the formatted messages in this case.  
+    ///             EnableEvents(eventSource, EventLevel.Error, LoggingEventSource.Keywords.FormattedMessage, args);
+    ///         }
+    ///     }
+    ///     protected override void OnEventWritten(EventWrittenEventArgs eventData) {
+    ///         // Look for the formatted message event, which has the following argument layout (as defined in the LoggingEventSource.  
+    ///         // FormattedMessage(LogLevel Level, int FactoryID, string LoggerName, string EventId, string FormattedMessage);
+    ///         if (eventData.EventName == "FormattedMessage")
+    ///             Console.WriteLine("Logger {0}: {1}", eventData.Payload[2], eventData.Payload[4]);
+    ///     }
+    /// }
+    /// </summary>
+    [EventSource(Name = "Microsoft-Extensions-Logging")]
+    public class LoggingEventSource : EventSource
+    {
+        /// <summary>
+        /// This is public from an EventSource consumer point of view, but since these defintions 
+        /// are not needed outside this class 
+        /// </summary>
+        public class Keywords
+        {
+            /// <summary>
+            /// Meta events are evnets about the LoggingEventSource itself (that is they did not come from ILogger
+            /// </summary>
+            public const EventKeywords Meta = (EventKeywords)1;
+            /// <summary>
+            /// Turns on the 'Message' event when ILogger.Log() is called.   It gives the information in a programatic (not formatted) way  
+            /// </summary>
+            public const EventKeywords Message = (EventKeywords)2;
+            /// <summary>
+            /// Turns on the 'FormatMessage' event when ILogger.Log() is called.  It gives the formatted string version of the information.  
+            /// </summary>
+            public const EventKeywords FormattedMessage = (EventKeywords)4;
+            /// <summary>
+            /// Turns on the 'MessageJson' event when ILogger.Log() is called.   It gives  JSON representation of the Arguments. 
+            /// </summary>
+            public const EventKeywords JsonMessage = (EventKeywords)8;
+        }
+
+        /// <summary>
+        ///  The one and only instance of the LoggingEventSource.  
+        /// </summary>
+        internal static readonly LoggingEventSource Instance = new LoggingEventSource();
+
+        internal static readonly LogLevel LoggingDisabled = LogLevel.None + 1;
+
+        private string _filterSpec;
+        private EventSourceLoggerProvider _loggingProviders;
+
+        internal EventSourceLoggerProvider CreateLoggerProvider(ILoggerFactory factory)
+        {
+            lock (this)
```

Comment:
It is a concern if the object that you are locking on is publicly available (because someone else might take a lock on it from outside, screwing up your logic). That is not the case here.